activity-monitor/activity-monitor.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2014 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"github.com/codegangsta/cli"
+	"github.com/letsencrypt/boulder"
+	"github.com/streadway/amqp"
+	"log"
+	"net/url"
+	"os"
+)
+
+const (
+	QueueName        = "Monitor"
+	AmqpExchange     = "boulder"
+	AmqpExchangeType = "topic"
+	AmqpInternal     = false
+	AmqpDurable      = false
+	AmqpDeleteUnused = false
+	AmqpExclusive    = false
+	AmqpNoWait       = false
+	AmqpNoLocal      = false
+	AmqpAutoAck      = true
+	AmqpMandatory    = false
+	AmqpImmediate    = false
+)
+
+
+func startMonitor(AmqpUrl string, logger *boulder.JsonLogger) {
+
+	conn, err   := amqp.Dial(AmqpUrl)
+	if err != nil {
+		log.Fatalf("Could not connect to AMQP server: %s", err)
+		return
+	}
+
+	rpcCh, err  := conn.Channel()
+	if err != nil {
+		log.Fatalf("Could not start channel: %s", err)
+		return
+	}
+
+	err = rpcCh.ExchangeDeclare(
+		AmqpExchange,
+		AmqpExchangeType,
+		AmqpDurable,
+		AmqpDeleteUnused,
+		AmqpInternal,
+		AmqpNoWait,
+		nil)
+	if err != nil {
+		log.Fatalf("Could not declare exchange: %s", err)
+		return
+	}
+
+	_, err = rpcCh.QueueDeclare(
+		QueueName,
+		AmqpDurable,
+		AmqpDeleteUnused,
+		AmqpExclusive,
+		AmqpNoWait,
+		nil)
+	if err != nil {
+		log.Fatalf("Could not declare queue: %s", err)
+		return
+	}
+
+	err = rpcCh.QueueBind(
+		QueueName,
+		"#", //wildcard
+		AmqpExchange,
+		false,
+		nil)
+	if err != nil {
+		log.Fatalf("Could not bind queue: %s", err)
+		return
+	}
+
+	delveries, err := rpcCh.Consume(
+		QueueName,
+		"", // dynamic queue name is fine
+		AmqpAutoAck,
```

Comment:
The activity monitor seems like a good place to disable auto-ack for reliability. We should instead ack once we've handed off the message successfully to jsonLogger, right?